examples/maps_heatmaps/src/index.ts

```diff
@@ -126,471 +120,204 @@ function getPoints() {
+  day=Math.floor(Math.random() * 29) + 1;
+  month=Math.floor(Math.random() * 12) + 1
+  year=Math.floor(Math.random() * 30) + 1990
+  return [day,month,year]
+}
+
+// add Single doc into database
+function addNewImage(label:string,lat: number, lng:number, year: number, month: number, day:number, url:string){
+  let newDoc = database.collection('imagesTal').doc();
+  newDoc.set({
+    year: year,
+    month: month, 
+    day: day,
+    url:url,
+    label: [label],
+    coordinates: new firebase.firestore.GeoPoint(lat, lng)  })
+  newDoc.collection('labels').doc().set({
+    name: label
+  })
+  }
+
+//add random data into the database 
+function addAllToDatabase()
+{
+  let arraydate:number[];
+  let randomnu: number;
+  for (let i = 0; i < allcoordinates.length; i++) {
+    arraydate=randomDate()
+    randomnu= Math.floor(Math.random() * 3)
+    if (randomnu===0)
+       addNewImage('dog',allcoordinates[i][0], allcoordinates[i][1],arraydate[2],arraydate[1],arraydate[0],'https://live.staticflickr.com/5284/5338762379_59f7435b93_c.jpg')
+    if (randomnu===1)
+      addNewImage('bag',allcoordinates[i][0], allcoordinates[i][1],arraydate[2],arraydate[1],arraydate[0],'https://live.staticflickr.com/65535/49748702651_07ae2b33b4_c.jpg')
+    if (randomnu===2)
+      addNewImage('cat',allcoordinates[i][0], allcoordinates[i][1],arraydate[2],arraydate[1],arraydate[0],'https://live.staticflickr.com/3677/13545844805_170ec3746b_c.jpg')
+  }
+}
+
+
+
+//******************************qeurying the database********************************************** */
+
+
+//a function than returns all the coordinates
+function queryAllData():Array<google.maps.LatLng>
+{
+  const allpoints : Array<google.maps.LatLng> = [];
+  const images = database.collection("imagesTal");
+  images.get().then(
+     queryImages => { queryImages.forEach(
+        doc=> {prinInfoOnObject(doc);
+          allpoints.push( nowlatlon( doc.get('coordinates').latitude, doc.get('coordinates').longitude ))})});
+    return allpoints  
+}
+
+//a function than returns fillter collection of  all the coordinates in a certain year
+function qeuryByYear(year:number)
+{
+  const allpoints : Array<google.maps.LatLng> = []; 
+  const dataRef:firebase.firestore.Query = database.collection('imagesTal');
+  return dataRef.where('year', '==', year);
+} 
+
+//a function returns a fillter collection that contain coordinates in a certain time from a certain label
+function queryByDateAndLabel(label:string,year?: number, month?: number, day?:number)
+{
+  const allpoints : Array<google.maps.LatLng> = []; 
+  let dataRef:firebase.firestore.Query = database.collection('imagesTal');
+  dataRef = dataRef.where('label', 'array-contains', label);
+  if (year != null)
+    dataRef = dataRef.where('year', '==', year);
+  if (year != null && month!=null)
+    dataRef = dataRef.where('month', '==', month)
+  if (year != null && month!=null && day!=null)
+    dataRef = dataRef.where('day', '==', day);
+  return dataRef;
+}
```

Comment:
Since we start to have a lot of query functions, please move all of them to a new file, e.g. (queries.ts)